data-products/src/sql_etl/sql/glean_firefox_new_tab_impressions_daily/glean_firefox_new_tab_daily_disable_rate_by_feed/data.sql

```diff
@@ -0,0 +1,95 @@
+{% set sql_engine = "bigquery" %}
+{% import 'helpers.j2' as helpers with context %}
+{% if for_new_offset %}
+    select current_timestamp()
+{% else %}
+
+--replicates logic of current Prefect query using Glean data
+--https://github.com/Pocket/data-flows/blob/main-v2/data-products/src/sql_etl/sql/firefox_new_tab_impressions_daily/firefox_new_tab_daily_disable_rate_by_feed/data.sql
+
+WITH
+  deduplicated_pings AS (
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_live.newtab_v1`
+  WHERE
+    submission_timestamp >= {{ helpers.parse_iso8601(batch_start) }}
+    AND submission_timestamp < {{ helpers.parse_iso8601(batch_end) }} QUALIFY ROW_NUMBER() OVER (PARTITION BY DATE(submission_timestamp),
+      document_id
+    ORDER BY
+      submission_timestamp DESC) = 1 )
+SELECT
+  DATE(submission_timestamp) AS happened_at,
+  CASE
+    WHEN ( normalized_country_code IN ('US', 'CA') AND metrics.string.newtab_locale IN ('en-CA', 'en-GB', 'en-US') ) THEN 'NEW_TAB_EN_US'
+    WHEN ( normalized_country_code IN ('GB',
+      'IE')
+    AND metrics.string.newtab_locale IN ('en-CA',
+      'en-GB',
+      'en-US') ) THEN 'NEW_TAB_EN_GB'
+    WHEN ( normalized_country_code IN ('IN') AND metrics.string.newtab_locale IN ('en-CA', 'en-GB', 'en-US') ) THEN 'NEW_TAB_EN_INTL'
+    WHEN ( normalized_country_code IN ('DE',
+      'CH',
+      'AT',
+      'BE')
+    AND metrics.string.newtab_locale IN ('de',
+      'de-AT',
+      'de-CH') ) THEN 'NEW_TAB_DE_DE'
+    WHEN (normalized_country_code IN ('IT') AND metrics.string.newtab_locale IN ('it')) THEN 'NEW_TAB_IT_IT'
+    WHEN (normalized_country_code IN ('FR')
+    AND metrics.string.newtab_locale IN ('fr')) THEN 'NEW_TAB_FR_FR'
+    WHEN (normalized_country_code IN ('ES') AND metrics.string.newtab_locale IN ('es-ES')) THEN 'NEW_TAB_ES_ES'
+END
+  AS feed_name,
```

Comment:
The feed name exists in Snowflake in `dbt_staging.stg_corpus_slate_recommendations.recommendation_surface_id`, to which this data can be joined using the `recommendation_id` that exists in `event.extra`. This data is reliable because it's directly based on the decision made by the recommendation API.

Maybe you could ingest the data from BigQuery without `feed_name` and then model `feed_name` in Dbt? That would remove duplicate SQL code and make `feed_name` more reliable. Given the complexity of this SQL code, I wouldn't be surprised if it doesn't perfectly match the actual feed.

Reply:
Hey Mathijs! That's right. You arrived at the conclusion yourself before I could reply 🏆  Some of these extraction queries are aggregated cuts of data without the tile ID or recommendation ID, so we can't do the mapping in Snowflake (and bringing over the raw data into Snowflake would be quite large).

On the models where we have tile/recommendation ID, we're not doing this mapping in the query itself and it will be done in dbt/Snowflake.

_Ideally_, Firefox could keep track of the feed name itself and track it directly in the Glean events!

Reply:
On second thought, this table only needs to be grouped by `recommendation_id` and `tile_id`, which is only ~700k rows per day. That could easily be loaded into Snowflake.